qiskit/chemistry/algorithms/minimum_eigen_solvers/oovqe.py

```diff
@@ -0,0 +1,604 @@
+                    optimizer=self.optimizer)
+                self.initial_point[self.var_form_num_parameters:] = vqresult.optimal_point
+        else:
+            # simultaneous method (ansatz and orbitals are optimized at the same time)
+            self.var_form._bounds = self.bounds
+            self.var_form._num_parameters = len(self.bounds)
+
+            vqresult = self.find_minimum(initial_point=self.initial_point,
+                                         var_form=self.var_form,
+                                         cost_fn=self._energy_evaluation_oo,
+                                         optimizer=self.optimizer)
+
+        # write original number of parameters to avoid errors due to parameter number mismatch
+        self.var_form._num_parameters = self.var_form_num_parameters
+
+        # save the results
+        self._ret = {}
+        self._ret['num_optimizer_evals'] = vqresult.optimizer_evals
+        self._ret['min_val'] = vqresult.optimal_value
+        if self.iterative_oo:
+            self._ret['opt_params'] = self.initial_point[self.var_form_num_parameters:]
+            self._ret['opt_params_orbitals'] = self.initial_point[:self.var_form_num_parameters]
+        else:
+            self._ret['opt_params'] = vqresult.optimal_point[:self.var_form_num_parameters]
+            self._ret['opt_params_orbitals'] = vqresult.optimal_point[self.var_form_num_parameters:]
+
+        self._ret['eval_time'] = vqresult.optimizer_time
+        self._ret['opt_psarams_dict'] = vqresult.optimal_parameters
+
+        if self._ret['num_optimizer_evals'] is not None and \
+                self._eval_count >= self._ret['num_optimizer_evals']:
+            self._eval_count = self._ret['num_optimizer_evals']
+        self._eval_time = self._ret['eval_time']
+        logger.info('Optimization complete in %s seconds.\nFound opt_params %s in %s evals',
+                    self._eval_time, self._ret['opt_params'], self._eval_count)
+        self._ret['eval_count'] = self._eval_count
+
+        result = VQEResult()
+        result.combine(vqresult)
+        result.eigenvalue = vqresult.optimal_value + 0j
+        # record all parameters (wavefunction and orbitals) to overcome error checks
+        _ret_temp_params = copy.copy(self._ret['opt_params'])
+        self._ret['opt_params'] = self._ret['opt_params'][:self.var_form_num_parameters]
+        result.eigenstate = self.get_optimal_vector()
+        self._ret['opt_params'] = _ret_temp_params
+
+        self._ret['energy'] = self.get_optimal_cost()
+        self._ret['eigvals'] = np.asarray([self._ret['energy']])
+        self._ret['eigvecs'] = np.asarray([result.eigenstate])
+
+        if self.aux_operators:
+            self._eval_aux_ops()
+            # TODO remove when ._ret is deprecated
+            result.aux_operator_eigenvalues = self._ret['aux_ops'][0]
+
+        result.cost_function_evals = self._eval_count
+
+        return result
+
+    def set_bounds(self,
```

Comment:
See earlier comment in regards if this being perhaps a private method for setting up the default. Similar for set_initial_point below

Reply:
Fixed